tests/loggen/logline_generator.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2007-2018 Balabit
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include <stdio.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <string.h>
+#include <glib.h>
+
+#include "logline_generator.h"
+#include "loggen.h"
+#include "loggen_helper.h"
+
+static char line_buf_template[MAX_MESSAGE_LENGTH + 1];
+static int pos_timestamp1 = 0;
+static int pos_timestamp2 = 0;
+static int pos_seq = 0;
+static int pos_thread_id = 0;
+
+int
+prepare_log_line_template(int syslog_proto, int framing, int message_length, char *sdata_value)
+{
+  int linelen = 0;
+  char padding[] = "PADD";
+  int hdr_len = 0;
+  struct timeval now;
+  gettimeofday(&now, NULL);
+  int run_id = now.tv_sec;
+
+  int buffer_length = sizeof(line_buf_template);
+
+  if (syslog_proto)
+    {
+      if (framing)
+        hdr_len = snprintf(line_buf_template, buffer_length, "%d ", message_length);
+      else
+        hdr_len = 0;
+
+      const char *sdata = sdata_value ? sdata_value : "-";
+
+      linelen = snprintf(line_buf_template + hdr_len, buffer_length - hdr_len,
+                         "<38>1 2007-12-24T12:28:51+02:00 localhost prg%05d 1234 - %s \xEF\xBB\xBFseq: %010d, thread: %04d, runid: %-10d, stamp: %-19s ",
+                         0, sdata, 0, 0, run_id, "");
+
+      pos_timestamp1 = 6 + hdr_len;
+      pos_seq = 68 + hdr_len + strlen(sdata) - 1;
+      pos_thread_id = pos_seq + 20;
+      pos_timestamp2 = 120 + hdr_len + strlen(sdata) - 1;
+    }
+  else
+    {
+      if (framing)
+        hdr_len = snprintf(line_buf_template, sizeof(line_buf_template), "%d ", message_length);
```

Comment:
This block is executed unconditionally to the if. Lets pull this out to a common place. The only difference between the two instances is sizeof(line_buf_template) is replaced with buffer_length

Reply:
done